src/com/facebook/buck/apple/AppleLibraryDescription.java

```diff
@@ -885,7 +885,7 @@ private static CxxHeaders createSwiftModuleHeaders(
     BuildTarget swiftCompileTarget = baseTarget.withAppendedFlavors(Type.SWIFT_COMPILE.getFlavor());
     SwiftCompile compile = (SwiftCompile) graphBuilder.requireRule(swiftCompileTarget);
 
-    return CxxHeadersDir.of(CxxPreprocessables.IncludeType.LOCAL, compile.getOutputPath());
+    return CxxHeadersDir.of(CxxPreprocessables.IncludeType.LOCAL, compile.getSwiftModuleOutputPath());
```

Comment:
So.. how this line will change rulekey calculation?

Reply:
@qyang-nj, I added a new section in the PR description to answer both of your questions.